src/functions/runtimeConfigExport.ts

```diff
@@ -0,0 +1,222 @@
+import * as fs from "fs";
+import * as path from "path";
+
+import * as clc from "cli-color";
+
+import * as env from "./env";
+import * as functionsConfig from "../functionsConfig";
+import { FirebaseError } from "../error";
+import { logger } from "../logger";
+import { getProjectId } from "../projectUtils";
+import { loadRC } from "../rc";
+import { logWarning } from "../utils";
+
+export interface ProjectConfigInfo {
+  projectId: string;
+  alias?: string;
+  config?: Record<string, unknown>;
+  envs?: EnvMap[];
+}
+
+export interface EnvMap {
+  origKey: string;
+  newKey: string;
+  value: string;
+  err?: string;
+}
+
+export interface ConfigToEnvResult {
+  success: EnvMap[];
+  errors: Required<EnvMap>[];
+}
+
+/**
+ * Find all projects (and its alias) associated with the current directory.
+ */
+export function getProjectInfos(options: {
+  project?: string;
+  projectId?: string;
+  cwd?: string;
+}): ProjectConfigInfo[] {
+  const result: Record<string, string> = {};
+
+  const rc = loadRC(options);
+  if (rc.projects) {
+    for (const [alias, projectId] of Object.entries(rc.projects)) {
+      if (Object.keys(result).includes(projectId)) {
+        logWarning(
+          `Multiple aliases found for ${clc.bold(projectId)}. ` +
+            `Preferring alias (${clc.bold(result[projectId])}) over (${clc.bold(alias)}).`
+        );
+        continue;
+      }
+      result[projectId] = alias;
+    }
+  }
+
+  const projectId = getProjectId(options);
```

Comment:
nit: can we get a comment here explaining that this detects the case where the customer said to `use` a particular project that they never added to their `.firebaserc` file?